### PR TITLE
Log level design -- a start

### PR DESCRIPTION
**Pull request by mih**
This is not comprehensive, but a mere start and invitation to
contribute.

Work towards datalad/datalad#5831

### CODE REVIEW

File: docs/source/design/log_levels.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_dataset_argument:
```
[the remark this answers is not in the document]
Comment (yarikoptic) [reply docs/source/design/log_levels.rst:4]: should may be it become a generic `Logging` document with for this PR a single subsection on `Log Levels`?

Later we could extend with other related information/conventions (exceptions logging, avoidance of explicit `%` and `.format`, etc) ?

File: docs/source/design/log_levels.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+=========
+
+Command execution
+-----------------
+
+For the non-async Runner and its protocols the following log levels are used:
+
+- High-level execution -> ``debug``
+- Process start/finish -> ``8``
+- Threading and IO -> ``5``
```
Comment (yarikoptic) [inline docs/source/design/log_levels.rst:32]: May be here add a section on related config/envvar (may be some envvars no longer in use) settings ?

<details>
<summary>what I see for config + envvars in CONTRIBUTING.md</summary> 

```shell
(git)lena:~datalad/datalad-master[master]git
$> git grep 'datalad\.log' datalad/interface/common_cfg.py
datalad/interface/common_cfg.py:    'datalad.log.level': {
datalad/interface/common_cfg.py:    'datalad.log.result-level': {
datalad/interface/common_cfg.py:    'datalad.log.name': {
datalad/interface/common_cfg.py:    'datalad.log.names': {
datalad/interface/common_cfg.py:    'datalad.log.namesre': {
datalad/interface/common_cfg.py:    'datalad.log.outputs': {
datalad/interface/common_cfg.py:    'datalad.log.timestamp': {
datalad/interface/common_cfg.py:    'datalad.log.traceback': {
datalad/interface/common_cfg.py:    'datalad.log.exc': {

$> grep LOG CONTRIBUT*                                    
CONTRIBUTING.md:- *DATALAD_LOG_LEVEL*:
CONTRIBUTING.md:- *DATALAD_LOG_NAME*:
CONTRIBUTING.md:- *DATALAD_LOG_OUTPUTS*:
CONTRIBUTING.md:- *DATALAD_LOG_PID*
CONTRIBUTING.md:- *DATALAD_LOG_TARGET*
CONTRIBUTING.md:- *DATALAD_LOG_TIMESTAMP*:
CONTRIBUTING.md:- *DATALAD_LOG_TRACEBACK*:
CONTRIBUTING.md:- *DATALAD_LOG_VMEM*:

```
</details>

**Review by yarikoptic — COMMENTED** [review]
initial review/comments

Thank you @mih!

File: docs/source/design/log_levels.rst [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+categorized into distinct levels. While some levels have self-explanatory
+descriptions (e.g. ``warning``, ``error``), others are less specific (e.g.
+``info``, ``debug``).
+
+Common principles
+=================
+
+Parenthical log message use the same level
+  When log messages are used to indicate the start and end of an operation,
+  both start and end message use the same log-level.
```
[the remark this answers is not in the document]
Comment (mih) [reply docs/source/design/log_levels.rst:20]: I have built this locally, and it is the desired description list.

Thanks for turning on the preview builds!

**Comment by mih** [comment]
I have made a range of adjustments (thx @yarikoptic) and importantly added markup to declare status and scope of the document.

I agree that this document could be extended into a general specification on logging. However, I neither have the energy nor the competence to approach this with the aim to actually get it done. Let's make baby steps.

File: docs/source/design/log_levels.rst (hunk 2 above)
Comment (mih) [reply docs/source/design/log_levels.rst:38]: (see main comment)

**Comment by mih** [comment]
OK, let's move.

**Pull request merged**